flink-connectors/flink-sql-connector-hive-1.2.2/src/main/resources/META-INF/NOTICE

```diff
@@ -8,6 +8,41 @@ This project bundles the following dependencies under the Apache Software Licens
 
 - org.apache.hive:hive-exec:1.2.2
 - org.apache.hive:hive-metastore:1.2.2
+- org.apache.hive:hive-common:1.2.2
+- org.apache.hive:hive-serde:1.2.2
+- org.apache.hive.shims:hive-shims-0.20S:1.2.2
+- org.apache.hive.shims:hive-shims-0.23:1.2.2
+- org.apache.hive.shims:hive-shims-common:1.2.2
+- org.apache.hive:spark-client:1.2.2
+- com.twitter:parquet-hadoop-bundle:1.6.0
+- org.apache.thrift:libthrift:0.9.2
 - org.apache.thrift:libfb303:0.9.2
 - org.apache.orc:orc-core:1.4.3
 - io.airlift:aircompressor:0.8
+- commons-lang:commons-lang:2.6
+- org.apache.commons:commons-lang3:3.1
+- org.apache.avro:avro:1.7.5
+- org.apache.avro:avro-mapred:1.7.5
+- com.googlecode.javaewah:JavaEWAH:0.3.2
+- org.iq80.snappy:snappy:0.2
+- org.codehaus.jackson:jackson-core-asl:1.9.2
+- org.codehaus.jackson:jackson-mapper-asl:1.9.2
+- com.google.guava:guava:14.0.1
+- net.sf.opencsv:opencsv:2.3
+- joda-time:joda-time:2.5
+- org.objenesis:objenesis:1.2
+
+This project bundles the following dependencies under the BSD license.
+See bundled license files for details.
+
+- com.esotericsoftware.kryo:kryo:2.22
+- org.jodd:jodd-core:3.5.2
+- javolution:javolution:5.5.1
+- com.google.protobuf:protobuf-java:2.5.0
+- com.esotericsoftware.minlog:minlog:1.2
+- com.esotericsoftware.reflectasm:reflectasm:1.07
+
+This project bundles the following dependencies under the JSON license.
```

Comment:
We can not bundle anything that is licensed under the JSON license, it is "Category X": https://www.apache.org/legal/resolved.html#category-x

Reply:
I confirmed that hive 1.2.2 depends on this: https://github.com/apache/hive/blob/rel/release-1.2.2/ql/pom.xml#L295

This basically means we have to drop hvie 1.2.2 support.

Reply:
We can only support version 2.2.0+; https://issues.apache.org/jira/browse/HIVE-15144.

Reply:
OK then I'll remove flink-sql-connector-hive 1.2.2.

BTW, I was also looking at [HIVE-15144](https://issues.apache.org/jira/browse/HIVE-15144) and it mentions that JSON license was moved to CatX as of 2016-11-03. Does that mean we're allowed to bundle JSON libraries released before that date?

Reply:
Alternatively, if we exclude JSON from the shaded jar, i.e. by filtering out `org/json/**` from hive-exec, will that solve the problem here?

Reply:
> Does that mean we're allowed to bundle JSON libraries released before that date?
If you were able to find a distro were the license is different sure, But I looked at the oldest release available in maven central, and it still had the problematic passage.

> Alternatively, if we exclude JSON from the shaded jar, i.e. by filtering out org/json/** from hive-exec, will that solve the problem here?
Well, yes and no. Ultimately, by virtue of containing a category-X dependency, hive-exec is also category-x. We could filter out json.org:json here, indeed. But what about all the other dependencies to hive-exec? Those cannot state that we only depend on the non-json stuff.
So if a user checks out the Flink source, and runs some hive example or whatnot, then this must not pull in json.org:json. And I don't believe this to be feasible.

Reply:
Maybe we can discuss each scenario that involves hive-exec:

1. User deploys Flink and adds flink-sql-connector-hive (or his/her own hive-exec) to connect with Hive.
2. User checkouts Flink source and tries out our hive connector tests.

I guess No. 1 is OK if we exclude JSON from `flink-sql-connector-hive`. But we cannot completely get rid of JSON in No. 2, because JSON was included in hive-exec so we cannot exclude it from hive-exec, and from Flink perspective, it's not even a transitive dependency.
If No. 2 is really a problem, maybe we have to officially drop support for Hive 1.x ( which is bad :( ). At least we cannot have the hive-1.x maven profiles for testing.

Reply:
A possible solution to No. 2 is to publish our own hive-exec 1.x. I think that's what Spark does, although their initial motivation is not about license problems.

Reply:
I believe we can keep Hive 1.x support in the source code, with the bad JSON dependency, **if we add a warning to the documentation**. 
The "flink-connector-kinesis" connector used to have a Category X license (Amazon Software License) as well. For that connector, we didn't provide a binary release, users had to compile it themselves.

Reply:
>I believe we can keep Hive 1.x support in the source code, with the bad JSON dependency, if we add a warning to the documentation.

We did not only did not provide a binary license, but also had an opt-in profile for including the module.
Technically this is also the case right now (since we have a dedicated profile for setting the version to hive 1.1.x), but IMO it's not distinct enough from the other profiles. It very much looks like "just another" a version switch.

With that in mind, we'd either need to splice out hive 1.1 from the existing hive-connector (not feasible for practical reason I suppose), have an opt-in profile for hive as a whole (likely not feasible since the entire table ecosystem is interlinked like a web), or make the hive 1.1 profile especially explicit (in the documentation and source).

In any case however nothing we release to maven central or bundle in the distribution must have a reference to hive 1.1.
This implies that users would have to recompile Flink against hive 1.1 to use it. Given the reported frictions with Flink users getting up and running in the Table/SQL APIs in regards to connectors, which is especially worse with Hive, this to me implies that it would be basically unusable. IIRC, the _current state_ was already described as such.
Hence why I referred to this as not being _feasible_. _Possible_, yes, but I daresay that it's not worth the effort.

Reply:
Let me clarify a little bit about the hive connector jars to make sure we're on the same page.

`flink-connector-hive` is the module containing all the Flink code needed to integrate with Hive. `flink-connector-hive` depends on Hive 2.3.4 (provided scope), but it also has some maven profiles to build and test with other Hive versions from 1.x to 3.x. These profiles are for **testing only**. The artifact we publish to maven doesn't bundle any Hive class and depends on Hive 2.3.4. Users don't have to recompile it to run with different Hive versions. Theoretically, we won't drop support for 1.x by just removing the 1.x maven profile. It's only that we don't have test coverage for that version.

We also have four `flink-sql-connector-hive` jars. These jars are only meant to make things easier for users to deploy dependencies in order to integrate with Hive. W/o these jars, users have to add `flink-connector-hive` as well as several other jars to connect to Hive. With these jars, users only need to pick one of them (according to their Hive version) and add to classpath. Each `flink-sql-connector-hive` jar contains hive-exec code. But with the filter added in this PR, it no longer contains JSON classes.

IIUC, the only problematic scenario is `flink-connector-hive` with 1.x profile, which pulls in the JSON library. What do you think?

Reply:
Fair enough about `flink-connector-hive`. If it is just for testing, then it should be fine.

I don't think it's that easy though for `flink-sql-connector-hive`. The reality is: this component cannot be used without a category-x dependency. Whether this dependency is explicit in the build files, or implied by the packaging, does not really matter, because there's no practical difference.
At the end of the day, the dependency _exists_. So we can't release this jar to maven central, or have it in the build process by default.

Reply:
OK. Then we don't have to drop support for 1.x. But we cannot publish `flink-sql-connector-hive` for 1.x. That means users using 1.x will have to go back to the old way to add dependencies.
@JingsongLi Does this sound acceptable to you?

Reply:
> The reality is: this component cannot be used without a category-x dependency.

Hi @zentol 
User can use `flink-sql-connector-hive` with this json category-x dependency.
json is used in hive 1.x for testing, tez, compile and some very limited places. These places actually we never touched. So without json in `flink-sql-connector-hive`, it is OK.

Reply:
@JingsongLi It is good to know that json is not necessarily required, but I'm not sure if it changes anything.
We'd be moving on _very_ thin ice by arguing "you can use this component, that we released, with hive 1.2, which is overall category-x because it includes json.org:json, but it's fine because it's not used in any code-path that we call into."
I have not seen such a distinction being made, hence why I'm cautious about it.
There's also the question how exactly it was verified that in _all_ code-paths we use we do not call into json.org:json, or how to ensure that this does not happen in the future.

Reply:
> There's also the question how exactly it was verified that in all code-paths we use we do not call into json.org:json, or how to ensure that this does not happen in the future.

json.org:json is excluded when we add hive-exec to our jar. It doesn't exist in the jar we publish. If any code path calls into json.org:json, it probably will hit exceptions complaining class not found.

Reply:
After thinking about it some more, I believe this is fine. @lirui-apache @JingsongLi Thank you for this discussion.

It _is_ an annoying special-case though; we can probably save ourselves some headaches by eventually dropping hive 1.X support.
(E.g., if a user scans the flink dependencies from the source, this could potentially get flagged, and then we have to explain this whole thing again)

Reply:
Hey @zentol , thanks for sharing your insights about this issue, very informative. Given that Hive 1.x is still used by quite some users, I'm afraid we can't drop support for it any time soon w/o hurting the adoption of the feature. We can revisit this topic if it really brings us too much headaches in the future.

Reply:
Thanks @lirui-apache and @zentol for your discussion.